laplace/__init__.py

```diff
@@ -8,10 +8,13 @@
 
 from laplace.baselaplace import BaseLaplace, ParametricLaplace, FullLaplace, KronLaplace, DiagLaplace
 from laplace.lllaplace import LLLaplace, FullLLLaplace, KronLLLaplace, DiagLLLaplace
+from laplace.subnetlaplace import SubnetLaplace, FullSubnetLaplace
 from laplace.laplace import Laplace
 
 __all__ = ['Laplace',  # direct access to all Laplace classes via unified interface
            'BaseLaplace', 'ParametricLaplace',  # base-class and its (first-level) subclasses
            'FullLaplace', 'KronLaplace', 'DiagLaplace',  # all-weights
            'LLLaplace',  # base-class last-layer
-           'FullLLLaplace', 'KronLLLaplace', 'DiagLLLaplace'] # last-layer
+           'FullLLLaplace', 'KronLLLaplace', 'DiagLLLaplace',  # last-layer
+           'SubnetLaplace',  # base-class subnetwork
+           'FullSubnetLaplace']  # subnetwork
```

Comment:
Why are two classes required? I think for now, the `FullSubnetLaplace` could just be the default and implement exactly the algorithm as in your paper. If you could elaborate a bit on what you plan to add as subclasses, this would help me understand this choice better. From what I understand, a KronSubnet wouldn't make sense/work and diagonal neither because the diagonal is used to find the optimal subnet, right?

Reply:
Yes, agreed, I removed the `SubnetLaplace` base class, making `FullSubnetLaplace` the only/default option.